client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +250,51 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let now = Instant::now();
+		let remaining_duration = if now > slot_info.ends_at {
+			Duration::from_millis(0)
+		} else {
+			slot_info.ends_at - now
+		};
+		let inherent_data = match slot_info
+			.create_inherent_data
+			.create_inherent_data()
+			.timeout(remaining_duration)
```

Comment:
Please. Use. Futures-timer Delay.

Reply:
What. is. up. with. those. dots?.

Reply:
Good question @eskimor :P I think my phone wanted to troll me :P